npm-publish.bash

```diff
@@ -14,36 +14,52 @@ function err() {
 }
 
 function main() {
+    msg "branch is ${TRAVIS_BRANCH}"
     local module_version=$1
+    local package="package.json"
+    local tmp_package="$package.tmp"
     if [[ ! $module_version ]]; then
         err "first parameter must be the version number of the module to publish"
         return 10
     fi
 
-    local package="package.json"
-    local tmp_package="$package.tmp"
     if ! jq --arg tag "$TRAVIS_TAG" '.version = $tag' "$package" > "$tmp_package"; then
         err "failed to set version in $package"
         return 1
     fi
     cp "$tmp_package" "$package"
-    if [[ $NPM_TOKEN ]]; then
+
+     # npm honors this
+    rm -f "$target/.gitignore"
+
+    if [[ $TRAVIS_TAG =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
         msg "Creating local .npmrc using API key from environment"
         if ! ( umask 077 && echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > "$HOME/.npmrc" ); then
             err "failed to create $HOME/.npmrc"
             return 1
         fi
-    else
-        msg "assuming your .npmrc is setup correctly for this project"
-    fi
-
-    # npm honors this
-    rm -f "$target/.gitignore"
+        if ! ( cd "$target" && npm publish --access=public ); then
+            err "failed to publish node module"
+            cat "$target/npm-debug.log"
+            return 1
+        fi
+    else [[ $TRAVIS_BRANCH == "master" ]]; then
+        msg "Creating local .npmrc from details generated in Artifactory..."
+        if [[ $ATOMIST_REPO_TOKEN  && $ATOMIST_REPO_USER ]]; then
+            if ! ( curl -u"${ATOMIST_REPO_USER}":"${ATOMIST_REPO_TOKEN}" "https://atomist.jfrog.io/atomist/api/npm/auth" >  "$HOME/.npmrc"); then
```

Comment:
The parentheses should not be necessary.

The quiet option should be used with curl and stderr should be redirected to `/dev/null` so the user name and password do not leak into the logs.